internal/webhook/external/helpers.go

```diff
@@ -0,0 +1,77 @@
+package external
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/api"
+	"github.com/akuity/kargo/internal/indexer"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+type refreshResult struct {
+	successes int
+	failures  int
+}
+
+func refreshWarehouses(
+	ctx context.Context,
+	c client.Client,
+	repoURL string,
+) (*refreshResult, error) {
+	logger := logging.LoggerFromContext(ctx)
+	var warehouses v1alpha1.WarehouseList
+	err := c.List(
+		ctx,
+		&warehouses,
+		client.MatchingFields{
+			indexer.WarehousesBySubscribedURLsField: repoURL,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list warehouses: %w", err)
+	}
```

Comment:
This is going to find all Warehouses (and then refresh them all).

Since these webhook receivers are Project-level (namespaced) configuration, we should be limiting the Warehouses here to just those associated with a specific Project.

(We will worry about receivers defined at the cluster-level later since that's going to require the introduction of something like a ClusterConfig CRD. Let's consider that out of scope for this PR.)